src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3207,6 +3213,57 @@ private static bool HasNonConstantInitializer(ArrayBuilder<ImmutableArray<FieldO
             AddInitializers(builder.StaticInitializers, staticInitializers);
         }
 
+        private void CollectFieldsFromGlobalDeconstruction(ArrayBuilder<Symbol> builder, VariableComponentSyntax variableComponent,
+            VariableComponentAssignmentSyntax assignment)
+        {
+            switch (variableComponent.Kind())
+            {
+                case SyntaxKind.TypedVariableComponent:
+                    var typed = (TypedVariableComponentSyntax)variableComponent;
+                    CollectFieldsFromGlobalDeconstruction(
+                        builder,
+                        typed.Designation,
+                        typed.Type,
+                        assignment);
+                    break;
+
+                case SyntaxKind.ParenthesizedVariableComponent:
+                    foreach (VariableComponentSyntax variable in ((ParenthesizedVariableComponentSyntax)variableComponent).Variables)
+                    {
+                        CollectFieldsFromGlobalDeconstruction(builder, variable, assignment);
+                    }
+                    break;
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(variableComponent.Kind());
+            }
+        }
+
+        private void CollectFieldsFromGlobalDeconstruction(ArrayBuilder<Symbol> builder, VariableDesignationSyntax designation,
+            TypeSyntax type, VariableComponentAssignmentSyntax assignment)
+        {
+            switch (designation.Kind())
+            {
+                case SyntaxKind.SingleVariableDesignation:
+                    var field = SourceMemberFieldSymbolFromDesignation.Create(this, (SingleVariableDesignationSyntax)designation,
+                        type, DeclarationModifiers.Private, null, assignment);
```

Comment:
>type, DeclarationModifiers.Private, null, assignment); [](start = 24, length = 54)

Minor thing. Usually, arguments are indented of off beginning of the invocation rather than of off beginning of the statement.